lib/model/recent_senders.dart

```diff
@@ -0,0 +1,189 @@
+import 'dart:math' as math;
+
+import '../api/model/model.dart';
+
+extension Max<T extends num> on Iterable<T> {
+  /// Finds the maximum number in an [Iterable].
+  ///
+  /// Returns null if the [Iterable] is empty.
+  T? max() => isNotEmpty ? reduce(math.max) : null;
+}
+
+extension PossibleIndex<T extends num> on List<T> {
+  /// The index of [element] in this sorted list.
+  ///
+  /// Uses binary search to find the index, so the list must be sorted,
+  /// otherwise the result is unpredictable.
+  ///
+  /// If [element] is found, returns its index (0-based).
+  ///
+  /// ```dart
+  /// final somePrimes = [11, 13, 17, 23, 27, 29];
+  /// final index = somePrimes.possibleIndexOf(17); // 2
+  /// ```
+  /// If [element] is not found, returns the possible index (1-based)
+  /// with a negative sign, where if the [element] is inserted,
+  /// the order of the list is maintained.
+  ///
+  /// ```dart
+  /// final somePrimes = [11, 13, 17, 23, 27, 29];
+  /// int index = somePrimes.possibleIndexOf(7); // -1
+  /// index = somePrimes.possibleIndexOf(19); // -4
+  /// ```
+  int possibleIndexOf(T element) {
```

Comment:
lib/model/algorithms.dart would probably be a better place for a function like this. The functions there aren't written as `extension`s, but I'm not sure if that's deliberate and worth emulating for this one.

Reply:
Moved it into there. Still went with the `extension` approach as it feels relatively smooth when using it. But if you think it's inconsistent with its surrounding code, please let me know and I will change it to a top-level function.

Reply:
For this specific case, the `List.lowerBound` method provides a solution to some extent as it returns the `index` where the `element` should be, but it does not give any clue if the element is in the list or not. But this method comes with that additional info as stated in its docs.